servo: 0-unstable-2025-07-30 -> 0.0.1-unstable-2025-10-20

mweinelt opened the pull request:
Attempt to update and unbreak servo, but it currently fails to build `mozjs_sys`, because their `build.rs` calls `cargo metadata` which fails, because it requires the `libz-rs`, but servo has locked with `libz-sys` instead.

- `mozjs-sys`, a dependency of `mozjs`, gets locked with `libz-sys` in https://github.com/servo/servo/blob/e6b604a276959c218591b463f5b3b2ec8c3bcaec/Cargo.lock
- The call to `cargo metadata` in https://github.com/servo/mozjs/blob/c37cfc3a84006ece60a191a0b8663b22f9b1d401/mozjs-sys/build.rs#L143 fails with

```
servo> warning: mozjs_sys@0.140.0-6: Failed to link pre-built archive by No such file or directory (os error 2). Building from source instead.
servo> error: failed to run custom build command for `mozjs_sys v0.140.0-6 (https://github.com/servo/mozjs#bbbbf315)`
servo> 
servo> Caused by:
servo>   process didn't exit successfully: `/build/source/target/release/build/mozjs_sys-e7bdff826a329509/build-script-build` (exit status: 101)
servo>   --- stdout
servo>   cargo:warning=Failed to link pre-built archive by No such file or directory (os error 2). Building from source instead.
servo>   cargo:rerun-if-env-changed=AR_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=AR_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_AR
servo>   cargo:rerun-if-env-changed=AR
servo>   cargo:rerun-if-env-changed=AS_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=AS_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_AS
servo>   cargo:rerun-if-env-changed=AS
servo>   cargo:rerun-if-env-changed=CC_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CC_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CC
servo>   cargo:rerun-if-env-changed=CC
servo>   cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CFLAGS
servo>   cargo:rerun-if-env-changed=CFLAGS
servo>   cargo:rerun-if-env-changed=CLANGFLAGS_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CLANGFLAGS_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CLANGFLAGS
servo>   cargo:rerun-if-env-changed=CLANGFLAGS
servo>   cargo:rerun-if-env-changed=CPP_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CPP_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CPP
servo>   cargo:rerun-if-env-changed=CPP
servo>   cargo:rerun-if-env-changed=CPPFLAGS_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CPPFLAGS_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CPPFLAGS
servo>   cargo:rerun-if-env-changed=CPPFLAGS
servo>   cargo:rerun-if-env-changed=CXX_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CXX_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CXX
servo>   cargo:rerun-if-env-changed=CXX
servo>   cargo:rerun-if-env-changed=CXXFLAGS_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CXXFLAGS
servo>   cargo:rerun-if-env-changed=CXXFLAGS
servo>   cargo:rerun-if-env-changed=READELF_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=READELF_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_READELF
servo>   cargo:rerun-if-env-changed=READELF
servo>   cargo:rerun-if-env-changed=OBJCOPY_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=OBJCOPY_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_OBJCOPY
servo>   cargo:rerun-if-env-changed=OBJCOPY
servo>   cargo:rerun-if-env-changed=WASI_SDK_PATH_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=WASI_SDK_PATH_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_WASI_SDK_PATH
servo>   cargo:rerun-if-env-changed=WASI_SDK_PATH
servo>   cargo:rerun-if-env-changed=CPPFLAGS_x86_64-unknown-linux-gnu
servo>   cargo:rerun-if-env-changed=CPPFLAGS_x86_64_unknown_linux_gnu
servo>   cargo:rerun-if-env-changed=TARGET_CPPFLAGS
servo>   cargo:rerun-if-env-changed=CPPFLAGS
servo> 
servo>   --- stderr
servo>   Trying to download prebuilt mozjs static library from Github Releases
servo> 
servo>   thread 'main' panicked at /build/servo-0.0.1-unstable-2025-10-20-vendor/mozjs_sys-0.140.0-6/build.rs:143:66:
servo>   called `Result::unwrap()` on an `Err` value: CargoMetadata { stderr: "error: no matching package found\nsearched package name: `libz-rs-sys`\nperhaps you meant:      libdbus-sys or libz-sys\nlocation searched: directory source `/build/servo-0.0.1-unstable-2025-10-20-vendor` (which is replacing registry `crates-io`)\nrequired by package `mozjs_sys v0.140.0-6 (/build/servo-0.0.1-unstable-2025-10-20-vendor/mozjs_sys-0.140.0-6)`\n" }
servo>   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
servo> warning: build failed, waiting for other jobs to finish...
```

or more concretely and expanded, unwrapping 

```
called `Result::unwrap()` on an `Err` value: CargoMetadata { 
  stderr: "error: no matching package found\nsearched package name: `libz-rs-sys`
  perhaps you meant:      libdbus-sys or libz-sys
  location searched: directory source `/build/servo-0.0.1-unstable-2025-10-20-vendor` (which is replacing registry `crates-io`)
  required by package `mozjs_sys v0.140.0-6 (/build/servo-0.0.1-unstable-2025-10-20-vendor/mozjs_sys-0.140.0-6)`
" }
````
exec() from cargo_metadata looks at the return code and returns an Error based on that:
- https://docs.rs/cargo_metadata/0.20.0/src/cargo_metadata/lib.rs.html#1122-1138

This is reproducible when calling `cargo metadata` in `${servo.cargoDeps}/mozjs_sys*`, but not when running it on `mozjs_sys` directory on the git clone.



<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt committed:
servo: 0-unstable-2025-07-30 -> 0.0.1-unstable-2025-10-22

https://github.com/servo/servo/compare/e6b604a276959c218591b463f5b3b2ec8c3bcaec...efc8c0f6647e542e35bac732585bc3aef6578028

Co-authored-by: Niklas Korz <niklas@niklaskorz.de>

eljamm commented on pkgs/by-name/se/servo/package.nix:
```diff
@@ -159,7 +173,6 @@ rustPlatform.buildRustPackage {
   };
```
Yes, but currently the update script will bump the version to `0-unstable-*`, so we should either modify `update.sh` to account for the new tag (`0.0.1`) or we can use the update script above, which will bump the version to `0.0.1-unstable-*`.

mweinelt replied:
Applied.

mweinelt committed:
servo: add ngi team as maintainers

Added with the expectation that they'll earn that role through regular
updates, reviews and responding to issues.

mweinelt commented on pkgs/by-name/se/servo/package.nix:
```diff
@@ -159,7 +173,6 @@ rustPlatform.buildRustPackage {
   };
 
   meta = {
```
And from there it went unmaintained for 4 months until I was fed up with it being outdated and created the autoupdater.

I'll add the team now, and we'll see how it goes. At least now you are aware of my concerns.

eljamm replied:
Thank you.
